activerecord/lib/active_record/type/serialized.rb

```diff
@@ -16,29 +16,20 @@ def initialize(subtype, coder)
       end
 
       def deserialize(value)
-        if default_value?(value)
-          value
-        else
-          coder.load(super)
-        end
+        coder.load(super)
       end
 
       def serialize(value)
-        return if value.nil?
-        unless default_value?(value)
-          super coder.dump(value)
-        end
+        return if null_equivalent?(value)
```

Comment:
IMHO, we can further simplify this part.
I believe the coder is responsible for treating deserialized `nil` as serialized `nil`. (This could be different between coders.)
Also, it is responsible for keeping `coder.dump(value) == nil` when `value == coder.load(nil)`.

Reply:
I have fixed this, and removed some ambiguity from the document.